src/cow-react/modules/limitOrders/hooks/useGetInitialPrice.ts

```diff
@@ -14,16 +14,12 @@ type PriceResult = number | Error | undefined
 async function requestPriceForCurrency(chainId: number | undefined, currency: Currency | null): Promise<PriceResult> {
```

Comment:
Just a suggestion if you think it improves readability, we could do something like this
Also in both scenarios we would return the price as 1 if its **not native** and there is no currencyAddress but I'm not sure that is safe, but then again I cant think of a scenario where the currency exists, its not native and it doesn't have an address.

```javascript
async function requestPriceForCurrency(chainId: number | undefined, currency: Currency | null): Promise<PriceResult> {
  const currencyAddress = getAddress(currency)
  const parsePrice = (price: number, currency: Currency) => price * 10 ** (DEFAULT_DECIMALS + getDecimals(currency))

  if (!chainId || !currency) {
    return
  }

  try {
    if (currency.isNative || !currencyAddress) {
      return parsePrice(1, currency)
    }

    const result = await getNativePrice(chainId, currencyAddress)

    if (!result) {
      throw new Error('Cannot parse initial price')
    }

    const price = parsePrice(result.price, currency)
    if (!price) {
      throw new Error('Cannot parse initial price')
    }

    return price
  } catch (error) {
    return error
  }
}
```

Reply:
Looks very beautiful! Thanks @nenadV91